component-catalog/src/Examples/Carousel.elm

```diff
@@ -192,47 +208,205 @@ example =
+                    (\id _ ->
+                        ( [ "{ id = " ++ String.fromInt id
+                          , ", slideHtml = Html.text \"" ++ String.fromInt (id + 1) ++ " slide\""
+                          , ", labelledBy = Carousel.LabelledByIdOfVisibleLabel (String.fromInt (id + 1) ++ \" of \" ++ String.fromInt settings.items)"
+                          , "}"
+                          ]
+                            |> String.join "\n          "
+                        , { id = id
+                          , slideHtml = Html.text (String.fromInt (id + 1))
+                          , labelledBy = Carousel.LabelledByIdOfVisibleLabel (String.fromInt (id + 1) ++ " of " ++ String.fromInt settings.items)
+                          }
+                        )
+                    )
+
+        { viewPreviousButton, viewNextButton, slides, containerAttributes } =
+            Carousel.viewWithPreviousAndNextControls
+                { selected = model.selected
+                , panels = List.map Tuple.second allItems
+                , viewPreviousButton =
+                    Html.button [ Events.onClick (FocusAndSelectItem { select = previousId, focus = Nothing }) ]
+                        [ Html.text "Previous" ]
+                , viewNextButton =
+                    Html.button [ Events.onClick (FocusAndSelectItem { select = nextId, focus = Nothing }) ]
+                        [ Html.text "Next" ]
+                , labelledBy = Carousel.LabelledByIdOfVisibleLabel "Items"
+                , role = Carousel.Group
+                }
+    in
+    ( [ moduleName ++ ".viewWithPreviousAndNextControls"
+      , "    { selected = " ++ String.fromInt model.selected
+      , "    , controlListStyles = " ++ Tuple.first settings.controlListStyles
+      , "    , panels =" ++ Code.listMultiline (List.map Tuple.first allItems) 2
+      , "    , viewPreviousButton = Html.button [ Events.onClick identity ] [ Html.text \"Previous\" ]"
+      , "    , viewNextButton = Html.button [ Events.onClick identity ] [ Html.text \"Next\" ]"
+      , "    , labelledBy = Carousel.LabelledByIdOfVisibleLabel \"Items\""
+      , "    , role = Carousel.Group"
+      , "    }"
+      , "    |> (\\{ viewPreviousButton, viewNextButton, slides, containerAttributes } -> section containerAttributes [ slides, controls, viewPreviousButton, viewNextButton ] )"
+      ]
+        |> String.join "\n"
+    , Html.div containerAttributes [ slides, viewPreviousButton, viewNextButton ]
+    )
+
+
+viewWithCombinedControls : State -> ( String, Html Msg )
+viewWithCombinedControls model =
+    let
+        settings =
+            Control.currentValue model.settings
+
+        previousId =
+            if model.selected - 1 >= 0 then
+                model.selected - 1
+
+            else
+                settings.items - 1
+
+        nextId =
+            if model.selected + 1 < settings.items then
+                model.selected + 1
+
+            else
+                0
```

Comment:
If possible, I think the "next" and "previous" selections should happen _inside_ the carousel component, rather than out here (for `viewWithCombinedControls` and `viewWithPreviousAndNextControls`)

Reply:
Do you think it's reasonable to accept a new field like:
```
    , onRotateCarousel: Prev/Next -> msg
```

Or do you have something else in mind?